watertap/property_models/unit_specific/activated_sludge/modified_asm2d_properties.py

```diff
@@ -442,10 +522,90 @@ def _TSS(self):
         )
 
         def rule_TSS(b):
-            return b.TSS == b.VSS + b.ISS
+            return b.TSS == b.conc_mass_comp["X_H"]
```

Comment:
Are you sure about this change? Ideally, the test would show no change between this and previous.

Reply:
Also, if we are confident about using mass concentration for X_H directly, I would opt for connecting this to TSS using `add_objective_reference` instead of creating a constraint.

Reply:
Sorry - I meant to undo this change. I had started to change it, but then realized TSS wasn't in the c-code. Also, I didn't bother looking at the tests yet since I knew I left out the optimization bit.